modules/core/src/test/java/com/google/refine/operations/RecipeTests.java

```diff
@@ -0,0 +1,199 @@
+    class ColumnRemovalOperation extends AbstractOperation {
+
+        final String columnName;
+
+        public ColumnRemovalOperation(String columnName) {
+            this.columnName = columnName;
+        }
+
+        @Override
+        public Optional<Set<String>> getColumnDependencies() {
+            return Optional.of(Set.of(columnName));
+        }
+
+        @Override
+        public Optional<ColumnsDiff> getColumnsDiff() {
+            return Optional.of(ColumnsDiff.builder().deleteColumn(columnName).build());
+        }
+    }
+
+    class ColumnRenameOperation extends AbstractOperation {
+
+        final String oldName;
+        final String newName;
+
+        public ColumnRenameOperation(String oldName, String newName) {
+            this.oldName = oldName;
+            this.newName = newName;
+        }
+
+        @Override
+        public Optional<Set<String>> getColumnDependencies() {
+            return Optional.of(Set.of(oldName));
+        }
+
+        @Override
+        public Optional<ColumnsDiff> getColumnsDiff() {
+            return Optional.of(ColumnsDiff.builder().deleteColumn(oldName).addColumn(newName, oldName).build());
+        }
+    }
+
+    class ColumnSplitOperation extends AbstractOperation {
+
+        final String columnName;
+
+        public ColumnSplitOperation(String columnName) {
+            this.columnName = columnName;
+        }
+
+        @Override
+        public Optional<Set<String>> getColumnDependencies() {
+            return Optional.of(Set.of(columnName));
+        }
+
+        @Override
+        public Optional<ColumnsDiff> getColumnsDiff() {
+            return Optional.empty();
+        }
+    }
+
+    class ColumnTransformOperation extends AbstractOperation {
+
+        final String columnName;
+
+        public ColumnTransformOperation(String columnName) {
+            this.columnName = columnName;
+        }
+
+        @Override
+        public Optional<Set<String>> getColumnDependencies() {
+            return Optional.of(Set.of(columnName));
+        }
+
+        @Override
+        public Optional<ColumnsDiff> getColumnsDiff() {
+            return Optional.of(ColumnsDiff.modifySingleColumn(columnName));
+        }
+    }
+
+    class OpaqueOperation extends AbstractOperation {
+
+        OpaqueOperation() {
+        }
+
+        @Override
+        public Optional<Set<String>> getColumnDependencies() {
+            return Optional.empty();
+        }
+
+        @Override
+        public Optional<ColumnsDiff> getColumnsDiff() {
+            return Optional.empty();
+        }
+    }
+
+    @Test
+    public void testDeserialize() throws Exception {
+        Recipe recipe = ParsingUtilities.mapper.readValue(json, Recipe.class);
+
+        assertEquals(recipe.getOperations().size(), 1);
+        assertEquals(recipe.getOperations().get(0).getOperationId(), "unknown");
+
+        TestUtils.isSerializedTo(recipe, json);
+    }
+
+    @Test
+    public void testValidate() {
```

Comment:
Just so I'm clear, this is intended for where we are testing to validate a set of operations in a Recipe? Looks like it.
Hmm, maybe we can rename it to `testValidateRecipeOperations` for clarity down the line?

Reply:
Maybe just `testValidateMethod` to make it clear that we're testing `Recipe::validate()`.